### PR TITLE
util: validate no leading zeroes7702

### PR DESCRIPTION
**Pull request by gabrocheleau**
This fixes issue #4057, where validateNoLeadingZeroes rejects [0] (single zero byte) because in strict RLP encoding, 0 should be an empty array []. However, for EIP-7702 authorization lists, chainId=0 (wildcard) and yParity=0 are valid values that may be encoded as 0x00.

- Added a new function validateNoLeadingZeroesAllowZero() in packages/util/src/bytes.ts that allows a single 0x00 byte but still rejects leading zeroes in multi-byte values
- Updated verifyAuthorizationList() in packages/tx/src/capabilities/eip7702.ts to use the new lenient validation for chainId and yParity, while keeping strict validation for r, s, and nonce
- Added tests to verify 0x00 is now accepted for chainId and yParity

### CODE REVIEW

**Comment by holgerd77** [comment]
Ok, there was already so much discussion around this in the original issue, will leave it to you guys without adding some half-informed opinion. Guess you guys knows that these 0 issues are tricky in general, also in respect to RLP/No-RLP, please also keep the API backwards-compatible.

**Comment by jochem-brouwer** [comment]
This is for the JSON input right? Not the RLP input? Because that should be rejected, see: https://github.com/ethereum/execution-specs/pull/1905

**Comment by jochem-brouwer** [comment]
I am now confused, this PR introduces the bug. On master this RLP throws, on this PR it does not. It should throw.

Add to tests this snippet (with correct imports)

```typescript
const txBytes = "0x04f8c101808007830186a09400000000000000000000000000000000000000008080c0f85cf85a009400000000000000000000000000000000000000018001a04ea0b08e1040c474a7716ba418a2577f030ba36f01af597d7cbbaad5d3f0e490a05eb7659b38d740bacf432bbfefcb437860a8e7455d765e70c94d01e4e35397ce80a0bd99ce90045c50413124e19b3f9c36c60be82836cd75f87493fcf44fa4718874a015b9b3cf77a5b0a6df3582305c856585cf9f0670ae524666de38e3a83f3b767a"


describe('[EOACode7702Transaction]', () => {
  it.only('decode txn', () => {
    const rlp = createEOACode7702TxFromRLP(hexToBytes(txBytes))
    console.log(rlp)
  })
  ```

It should fail. I am now confused what case #4057 was talking about because it seems the issue is not correct.

NOTE: above tests is taken from https://github.com/ethereum/execution-specs/pull/1905

File: packages/tx/src/capabilities/eip7702.ts [hunk 1]
```diff
@@ -49,7 +50,10 @@ export function verifyAuthorizationList(tx: EIP7702CompatibleTx) {
 
     const [chainId, address, nonce, yParity, r, s] = item
 
-    validateNoLeadingZeroes({ yParity, r, s, nonce, chainId })
+    // For chainId and yParity, we allow 0x00 (single zero byte) as valid encoding for zero
+    // See: https://github.com/ethereumjs/ethereumjs-monorepo/issues/4057
+    validateNoLeadingZeroesAllowZero({ chainId, yParity })
```
Comment (jochem-brouwer) [inline packages/tx/src/capabilities/eip7702.ts:55]: I think this is incorrect, and maybe the issue itself is incorrect. The 0x00 encoding is not allowed on the RLP. There are multiple reasons:

- If you strictly would sign the authority list, it would yield different bytes to sign, so you need a new signature for this.
- If we would do the opposite (ignore 0x00 and convert it to 0x, but allow it for transport) then if we would strictly use it in the transaction to hash it, it would yield a different hash. So you have the "same" transaction with different hashes, which is not good

I think we should recover what issue #4057 was talking about. This PR: https://github.com/ethereum/execution-specs/pull/1905 just got merged and we would fail these transaction tests if we would merge these changes.

How should we continue? We could also import the raw tests here to ensure the 0x00 encoding of zero values are rejected directly in the tests (without waiting for the exec tests release)

**Review by jochem-brouwer — CHANGES_REQUESTED** [review]

**Comment by gabrocheleau** [comment]
Closing in favor of #4209
